### DISCUSSION TITLE
[docs] How to index places in a pmtile file and search them?

### DISCUSSION THREAD

**Discussion by contrun** [protomaps/PMTiles#310] [Q&A] [answered]
Image that I want to host the whole map data from [Builds | Protomaps Basemaps](https://maps.protomaps.com/builds/) with PMTiles, and users want to search places available from these pmtiles, how do I index all the places that are available in my pmtiles file and implement the search functionality?

I think this is a fairly common feature. Is it documented somewhere, if not, can we document this?

**Comment by bdon** [comment, marked as answer]
You need a geocoder like https://pelias.io - that is not possible to do correctly with visual tiles. 

(moved to discussions)

**Reply by contrun** [reply]
Thank you for your quick response. By "that is not possible to do correctly with visual tiles", do you mean that it is not possible to search places with pmtiles files alone (I don't expect pmtiles alone can do this)? You do mean that https://pelias.io etc can produce valid search results, right?

**Reply by bdon** [reply]
Yes, Pelias is a totally separate project for geocoding based on OSM and other data that is designed for search experiences.

**Reply by contrun** [reply]
Great.Thank you.
